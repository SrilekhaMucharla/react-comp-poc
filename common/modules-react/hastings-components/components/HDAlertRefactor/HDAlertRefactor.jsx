import React from 'react';
import PropTypes from 'prop-types';


const HDAlertRefactor = ({ message, className }) => ((message) ? (
    <div className={`alert-container ${className}`}>
        {/* <i className="fas fa-exclamation-circle" /> */}
        {message}
    </div>
) : null);

HDAlertRefactor.propTypes = {
    message: PropTypes.string,
    className: PropTypes.string
};

HDAlertRefactor.defaultProps = {
    message: null,
    className: ''
};

export default HDAlertRefactor;
