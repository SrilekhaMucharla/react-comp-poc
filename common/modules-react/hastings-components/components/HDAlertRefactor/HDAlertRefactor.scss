.alert-container {
  background: $error-color;
  border-radius: 6px;
  padding: 20px 25px;
  margin: 14px 0 30px;
  display: flex;
  color: $white-color;
  font-weight: 500;
  line-height: 1.5;
  i {
    padding-top: 2px;
    padding-right: 5px;
  }
}
