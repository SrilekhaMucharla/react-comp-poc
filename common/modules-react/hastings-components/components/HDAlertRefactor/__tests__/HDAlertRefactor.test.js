import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import HDAlertRefactor from '../HDAlertRefactor';

Enzyme.configure({ adapter: new Adapter() });

describe('<HDAlertRefactor />', () => {
    it('render component with message', () => {
        const wrapper = shallow(<HDAlertRefactor message="Error" />);
        expect(wrapper).toMatchSnapshot();
    });

    it('render component with message', () => {
        const wrapper = shallow(<HDAlertRefactor />);
        expect(wrapper).toMatchSnapshot();
    });
});
