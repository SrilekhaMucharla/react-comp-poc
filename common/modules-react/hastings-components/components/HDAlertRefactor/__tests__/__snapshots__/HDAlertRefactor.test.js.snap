// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HDAlertRefactor /> render component with message 1`] = `
<div
  className="alert-container "
>
  Error
</div>
`;

exports[`<HDAlertRefactor /> render component with message 2`] = `""`;
